symbolic-base/src/ZkFold/Symbolic/Data/JWT/Twitch.hs

```diff
@@ -0,0 +1,138 @@
+        , twIsUnlinked   :: VarByteString 40 ctx
+        -- ^ true when the token is for a user that previously shared identity; otherwise, false.
+        , twOpaqueUserId :: VarByteString 512 ctx
+        -- ^ Identifies the session using this JWT.
+        , twPubsubPerms  :: PubSubPerms ctx
+        -- ^ Defines the ability of the token holder to send and listen to messages for your extension.
+        , twRole         :: VarByteString 88 ctx
+        -- ^ Type of user for whom this JWT has been signed. This is required.
+        , twUserId       :: VarByteString 512 ctx
+        -- ^ The user's Twitch user ID. This is provided only for users who allow your extension to identify them.
+        }
+    deriving Generic
+
+-- | @pListen@ and @pSend@ contain the topics the associated user is allowed to listen to and publish to, respectively.
+--
+data PubSubPerms ctx =
+    PubSubPerms
+        { pListen :: VarByteString 2048 ctx
+        , pSend   :: VarByteString 2048 ctx
+        }
+    deriving Generic
+
+deriving instance
+    ( P.Eq (ctx (V.Vector 2048))
+    , P.Eq (ctx Par1)
+    ) => P.Eq (PubSubPerms ctx)
+deriving instance
+    ( P.Show (ctx (V.Vector 2048))
+    , P.Show (ctx Par1)
+    ) => P.Show (PubSubPerms ctx)
+deriving instance Symbolic ctx => SymbolicData (PubSubPerms ctx)
+deriving instance Symbolic ctx => SymbolicInput (PubSubPerms ctx)
+instance Symbolic ctx => Arbitrary (PubSubPerms ctx) where
+    arbitrary = genericArbitrary uniform
+
+instance (Symbolic ctx, Context (PubSubPerms ctx) ~ ctx) => IsSymbolicJSON (PubSubPerms ctx) where
+    type MaxLength (PubSubPerms ctx) = 4248
+    toJsonBits PubSubPerms{..} =
+                    (fromType @"{\"listen\":") @+ pListen
+        `VB.append` (fromType @",\"send\":")   @+ pSend
+        `VB.append` (fromType @"}")
+
+instance Symbolic ctx => FromJSON (PubSubPerms ctx) where
+    parseJSON = genericParseJSON (aesonPrefix snakeCase) . stringify
+
+deriving instance
+    ( P.Eq (ctx (V.Vector 40))
+    , P.Eq (ctx (V.Vector 80))
+    , P.Eq (ctx (V.Vector 88))
+    , P.Eq (ctx (V.Vector 512))
+    , P.Eq (ctx (V.Vector 2048))
+    , P.Eq (ctx Par1)
+    ) => P.Eq (TwitchPayload ctx)
+
+deriving instance
+    ( P.Show (ctx (V.Vector 40))
+    , P.Show (ctx (V.Vector 80))
+    , P.Show (ctx (V.Vector 88))
+    , P.Show (ctx (V.Vector 512))
+    , P.Show (ctx (V.Vector 2048))
+    , P.Show (ctx Par1)
+    ) => P.Show (TwitchPayload ctx)
+deriving instance Symbolic ctx => SymbolicData (TwitchPayload ctx)
+deriving instance Symbolic ctx => SymbolicInput (TwitchPayload ctx)
+instance Symbolic ctx => Arbitrary (TwitchPayload ctx) where
+    arbitrary = genericArbitrary uniform
+
+instance Symbolic ctx => FromJSON (TwitchPayload ctx) where
+    parseJSON = genericParseJSON (aesonPrefix snakeCase) . stringify
+
+instance (Symbolic ctx, Context (TwitchPayload ctx) ~ ctx) => IsSymbolicJSON (TwitchPayload ctx) where
```

Comment:
Why is the `Context (TwitchPayload ctx) ~ ctx` constraint needed here?

Reply:
Because `IsSymbolicJSON a` has a function from `a` to `VarByteString n (Context a)`. It does not know that `ctx` is, in fact, `Context (TwitchPayload ctx)`

Reply:
That's really strange that GHC cannot compute `Context (TwitchPayload ctx)` and unify it with `ctx`, could you please provide the error it reports here for me so I could take a look?

Reply:
Oh wow, GHC actually did infer the type equality! But I saw this error somewhere, I swear... Thank you!